pkg/agent/clientset.go

```diff
@@ -171,17 +167,17 @@ func (cs *ClientSet) syncOnce() error {
 	if cs.serverCount != 0 && cs.ClientsCount() >= cs.serverCount {
 		return nil
 	}
-	c, err := cs.newAgentClient()
+	c, serverCount, err := cs.newAgentClient()
 	if err != nil {
 		return err
 	}
-	cs.serverCount = c.stream.serverCount
-	if err := cs.AddClient(c.stream.serverID, c); err != nil {
+	cs.serverCount = serverCount
```

Comment:
I think we need to log a warning if cs.serverCount !=0 and cs.serverCount != serverCount. Whether we are looking at a serverCount of zero creeping in, two proxy servers/apiservers with differing beliefs in the needed number of servers or a legitimate runtime change in the count, it will be hard to debug these scenarios without this information. In addition this should be a very rare event.

Reply:
Good call. Updated.